src/server/pfs/server/driver.go

```diff
@@ -2109,6 +2109,27 @@ func (d *driver) addBranchProvenance(txnCtx *txncontext.TransactionContext, bran
 	return errors.EnsureStack(err)
 }
 
+func (d *driver) deleteProjectRepos(ctx context.Context, project *pfs.Project) error {
+	var repos []*pfs.Repo
+	if err := d.listRepo(ctx, !includeAuth, "", func(repoInfo *pfs.RepoInfo) error {
+		if repoInfo.GetRepo().GetProject().GetName() == project.GetName() {
+			repos = append(repos, repoInfo.Repo)
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+	return d.txnEnv.WithWriteContext(ctx, func(txnCtx *txncontext.TransactionContext) error {
+		// the list does not use the transaction
+		for _, repo := range repos {
+			if err := d.deleteRepo(txnCtx, repo, true); err != nil && !auth.IsErrNotAuthorized(err) {
```

Comment:
A bit out of scope, but I kinda prefer `errors.As(err, &auth.ErrNotAuthorized{})`, since `auth.IsErrNotAuthorized(err)` is using string matching. However, maybe we ought to leave this work to the overall error handling improvement work planned.